.github/workflows/build_and_test.yml

```diff
@@ -14,6 +14,11 @@ on:
     branches:
     - main
 
+# Cancel previous PR build and test
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.event.number || github.sha }}
```

Comment:
Is it enough to isolate `Pull Request`, @guiyanakuang ?

Reply:
I think it's enough, almost all contributions are currently committed by pull request. Also when we use to receive other events, we can degrade to grouping using `github.sha`.